src/tls.c

```diff
@@ -117,9 +117,12 @@ int mg_tls_init(struct mg_connection *c, struct mg_tls_opts *opts) {
     mbedtls_ssl_conf_ca_chain(&tls->conf, &tls->ca, NULL);
 #endif
     if (opts->srvname.len > 0) {
-      char buf[opts->srvname.len + 1];
-      sprintf(buf, "%.*s", (int) opts->srvname.len, opts->srvname.ptr);
-      mbedtls_ssl_set_hostname(&tls->ssl, buf);
+      char *buf = (char *) calloc(opts->srvname.len + 1, sizeof(char));
```

Comment:
Please use  `mg_vasprintf`:

```c
char mem[128], *buf = mem;
mg_vasprintf(&buf, sizeof(mem), ...);
...
if (buf != mem) free(buf);
```

Also, please sign a CLA: https://cesanta.com/cla.html
Make sure to squash all commits into one, in order to have only one commit per PR.

Thanks!

Reply:
updated as per your recommendation (had to use mg_asprintf though). I think I already signed a CLA.